.github/workflows/e2e-tests.yml

```diff
@@ -7,14 +7,31 @@ on:
       - 'release-**'
   pull_request:
     types: [opened, synchronize, reopened, ready_for_review]
+  workflow_run:
+    workflows: [External Trigger Filter]
+    types: [completed]
 
 concurrency:
   group: ${{ github.workflow }}-${{ github.head_ref && github.ref || github.run_id }}
   cancel-in-progress: true
 
 jobs:
+  external-filter:
+    if: github.event.pull_request.head.repo.full_name == github.repository || github.event.workflow_run.conclusion == 'success'
+    runs-on: ubuntu-22.04
+    timeout-minutes: 3
+    steps:
+      - run: |
+          if ${{ github.event.pull_request.head.repo.full_name == github.repository }}; then
+            echo 'Running E2E tests within the organization.'
+          elif ${{ github.event.workflow_run.conclusion == 'success' }}; then
+            echo "Running E2E tests for the external contributor. Thank you!"
+          fi
+
   files-changed:
     name: Check which files changed
+    needs: external-filter
```

Comment:
In theory, only these two, top-level, jobs need the conditional logic because all other jobs sequentially depend on them, and cannot execute if these two are skipped.

But as a safety measure, I don't think it hurts to add the same condition to every single job in a workflow.